moped-editor/src/views/projects/projectView/KnackSync.js

```diff
@@ -0,0 +1,217 @@
+     */
+    const getHttpMethod = () => {
+      let method = "POST";
+      if (project.moped_project[0].knack_project_id) {
+        method = "PUT";
+      }
+      return method;
+    };
+
+    /**
+     * Function to build up the headers which need to be sent as part of the a Knack API call
+     * @returns object
+     */
+    const buildHeaders = () => {
+      let headers = {
+        "Content-Type": "application/json",
+        "X-Knack-Application-Id": knackApplicationId,
+        "X-Knack-REST-API-Key": "knack",
+      };
+      return headers;
+    };
+
+    /**
+     * Function to build up a JSON object of the fields which need to be updated in a call to Knack. This is needed
+     * because if you update the project number field, even with the same, extant number, Knack returns an error.
+     * @returns string
+     */
+    const buildBody = () => {
+      let body = {};
+
+      let field_map = {
+        field_3998: "project_id",
+        field_3999: "project_name",
+        field_4000: "current_status",
+      };
+
+      Object.keys(field_map).forEach(element => {
+        if (
+          project.moped_project[0].currentKnackState[element] !==
+          project.moped_project[0][field_map[element]]
+        ) {
+          body[element] = project.moped_project[0][field_map[element]];
+        }
+      });
+
+      return JSON.stringify(body);
+    };
+
+    const [mutateProjectKnackId] = useMutation(UPDATE_PROJECT_KNACK_ID);
+
+    /**
+     * Function to hanlde the actual mechanics of syncronizing the data on hand to the Knack API endpoint.
+     */
+    const handleSync = () => {
+      if (project.moped_project[0].knack_project_id) {
+        // updating knack record
+        fetch(buildUrl(), {
+          method: "GET",
+          headers: buildHeaders(),
+        })
+          .then(response => response.json())
+          .then(
+            result => {
+              if (result.errors) {
+                // knack error
+                snackbarHandler({
+                  severity: "warning",
+                  message: "Success: Data Tracker sync failed.",
+                });
+                return Promise.reject(result);
+              } else {
+                // get-state success
+                project.moped_project[0].currentKnackState = result;
+                return fetch(buildUrl(), {
+                  method: getHttpMethod(),
+                  headers: buildHeaders(),
+                  body: buildBody(),
+                });
+              }
+            },
+            error => {
+              // get-state fetch error
+              return Promise.reject(error);
+            }
+          )
+          .then(response => response.json())
+          .then(
+            result => {
+              if (result.errors) {
+                // knack error
+                snackbarHandler({
+                  severity: "warning",
+                  message: "Success: Data Tracker sync failed.",
```

Comment:
```suggestion
                  message: "Error: Data Tracker sync failed.",
```

Reply:
I won't comment on all of these to avoid flooding the PR, but just a straightforward typo here on the error messaging.

Reply:
![Screen Shot 2021-11-17 at 12 47 46 PM](https://user-images.githubusercontent.com/5697474/142263369-be1aff2a-d908-42a9-bae1-06a0c22b0a60.png)

Reply:
Thanks for pointing that out! I've aligned the wording to the error states, added a missing snackbar error and made sure that all sentences have a `.` at the end of them.